plugins/mongo_db_plugin/mongo_db_plugin.cpp

```diff
@@ -292,15 +291,15 @@ bool mongo_db_plugin_impl::filter_include( const transaction& trx ) const
 
 template<typename Queue, typename Entry>
 void mongo_db_plugin_impl::queue( Queue& queue, const Entry& e ) {
-   boost::mutex::scoped_lock lock( mtx );
+   std::unique_lock<std::mutex> lock( mtx );
```

Comment:
This could use a double check

Reply:
`std::lock_guard` would be slightly better. But probably not worth pushing a new commit.

Reply:
I don't think `std::lock_guard` works here because the code then does things like `lock.unlock();` -- `unlock()/lock()` isn't exposed by lock_guard AFAICT

Reply:
You are correct. I missed the `unlock()/lock()`